ir/src/main/scala/org/scalajs/ir/Serializers.scala

```diff
@@ -44,10 +46,46 @@ object Serializers {
     new Deserializer(stream).deserializeEntryPointsInfo()
   }
 
+  def deserializeEntryPointsInfo(mkBuf: Int => ByteBuffer,
+      readMore: ByteBuffer => Future[Unit])(
```

Comment:
TBH, I really dislike this interface, but IDK what to do better (without introducing duplication).

Reply:
Couldn't we just take a `ByteBuffer` (like `deserialize`) and let the `BufferUnderflowException` propagate, documenting that if it is thrown, the deserialization should be attempted again from scratch, with more data?

Callers that have an already complete `ByteBuffer` wouldn't have anything to do (e.g., in-memory, which includes the in-memory extraction of jar contents). And callers who try to be smart about reading the raw data in the first place can do so in the most appropriate way.